src/lib/coda_lib/coda_lib.ml

```diff
@@ -595,16 +541,38 @@ let create (config : Config.t) =
           let proposer_transition_reader, proposer_transition_writer =
             Strict_pipe.create Synchronous
           in
-          let%bind ledger_db, transition_frontier =
-            create_genesis_frontier config ~verifier
+          (* TODO: (#3053) push transition frontier ownership down into transition router
+             * (then persistent root can be owned by transition frontier only) *)
+          let persistent_frontier =
+            Transition_frontier.Persistent_frontier.create
+              ~logger:config.logger ~verifier
+              ~time_controller:config.time_controller
+              ~directory:config.persistent_frontier_location
           in
-          let genesis_transition =
-            Transition_frontier.(
-              root transition_frontier |> Breadcrumb.validated_transition
-              |> External_transition.Validation.forget_validation)
+          let persistent_root =
+            Transition_frontier.Persistent_root.create ~logger:config.logger
+              ~directory:config.persistent_root_location
+          in
+          let%bind transition_frontier_opt =
+            Transition_frontier.load ~logger:config.logger ~verifier
+              ~consensus_local_state:config.consensus_local_state
+              ~persistent_root ~persistent_frontier ()
+            >>| function
+            | Ok frontier ->
+                Some frontier
+            | Error `Persistent_frontier_malformed ->
+                failwith
+                  "persistent frontier unexpectedly malformed -- this should \
+                   not happen with retry enabled"
```

Comment:
There's an optional argument to the function that sets whether or not you want to retry once. I figured that we would want to tie this into the CLI so that users could specify whether or not the deamon is allowed to automatically delete their local persistent frontier database. Since that hasn't been decided nor implemented yet, I will just give a generally useful message such as "try deleting your persistent frontier and try again" (which an example command to do exactly that).